presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/HiveMetastoreCommitHandle.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import com.facebook.presto.spi.connector.ConnectorCommitHandle;
+import org.joda.time.DateTime;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class HiveMetastoreCommitHandle
+        implements ConnectorCommitHandle
+{
+    private final Map<String, List<DateTime>> lastDataCommitTimes = new HashMap<>();
```

Comment:
We finalize all the entries through constructor. This will be an immutable map

Reply:
So we need a copy constructor for updating.